pkg/chunk/storage/factory.go

```diff
@@ -32,6 +32,21 @@ const (
 	StorageEngineBlocks = "blocks"
 )
 
+// Supported storage clients
+const (
+	AWS            = "aws"
+	Azure          = "azure"
+	Cassandra      = "cassandra"
+	InMemory       = "inmemory"
+	BigTable       = "bigtable"
+	BigTableHashed = "bigtable-hashed"
+	FileSystem     = "filesystem"
+	GCP            = "gcp"
+	GCS            = "gcs"
+	GrpcStore      = "grpc-store"
+	S3             = "s3"
```

Comment:
**suggestion(optional):** for enums like this, I think it makes sense to prefix them with some kind of consistent identifier. Out of context (like in an editor autocomplete dropdown) these variable names could be confusing but with a consistent prefix you know they are related. The other option is to make a custom enum type on top of the string type, [this site shows an example using an int](https://www.educative.io/edpresso/what-is-an-enum-in-golang).

```suggestion
	StoreTypeAWS            = "aws"
	StoreTypeAzure          = "azure"
	StoreTypeCassandra      = "cassandra"
	StoreTypeInMemory       = "inmemory"
	StoreTypeBigTable       = "bigtable"
	StoreTypeBigTableHashed = "bigtable-hashed"
	StoreTypeFileSystem     = "filesystem"
	StoreTypeGCP            = "gcp"
	StoreTypeGCS            = "gcs"
	StoreTypeGrpc      = "grpc-store"
	StoreTypeS3             = "s3"
```

Reply:
I added the prefix `StorageType` to follow the style here: https://github.com/cortexproject/cortex/blob/cb8ed0e4eaaa73f91881a13e57c014bfffa468a2/pkg/chunk/storage/factory.go#L30-L33